triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerRuleSimulationLib.scala

```diff
@@ -6,93 +6,652 @@ package com.daml.lf.engine.trigger
+          commands = metricCountData(uuid)("in-flight-start")
+        ),
+      ),
+      endState = InternalStateMetrics(
+        acs = ACSMetrics(
+          activeContracts = metricCountData(uuid)("acs-active-end"),
+          pendingContracts = metricCountData(uuid)("acs-pending-end"),
+        ),
+        inFlight = InFlightMetrics(
+          commands = metricCountData(uuid)("in-flight-end")
+        ),
+      ),
+    )
+  }
+
+  private def addACSActiveStart(logEntry: JsObject): Unit = {
+    try {
+      require("Trigger rule .+ start".r.matches(getMessage(logEntry)))
+
+      metricCountData(getSpanId(logEntry)) += ("acs-active-start" -> getACSActive(logEntry))
+    } catch {
+      case NonFatal(_) =>
+    }
+  }
+
+  private def addACSActiveEnd(logEntry: JsObject): Unit = {
+    try {
+      require("Trigger rule .+ end".r.matches(getMessage(logEntry)))
+
+      metricCountData(getSpanId(logEntry)) += ("acs-active-end" -> getACSActive(logEntry))
+    } catch {
+      case NonFatal(_) =>
+    }
+  }
+
+  private def addACSPendingStart(logEntry: JsObject): Unit = {
+    try {
+      require("Trigger rule .+ start".r.matches(getMessage(logEntry)))
+
+      metricCountData(getSpanId(logEntry)) += ("acs-pending-start" -> getACSPending(logEntry))
+    } catch {
+      case NonFatal(_) =>
+    }
+  }
+
+  private def addACSPendingEnd(logEntry: JsObject): Unit = {
+    try {
+      require("Trigger rule .+ end".r.matches(getMessage(logEntry)))
+
+      metricCountData(getSpanId(logEntry)) += ("acs-pending-end" -> getACSPending(logEntry))
+    } catch {
+      case NonFatal(_) =>
+    }
+  }
+
+  private def addInFlightStart(logEntry: JsObject): Unit = {
+    try {
+      require("Trigger rule .+ start".r.matches(getMessage(logEntry)))
+
+      metricCountData(getSpanId(logEntry)) += ("in-flight-start" -> getInFlight(logEntry))
+    } catch {
+      case NonFatal(_) =>
+    }
+  }
+
+  private def addInFlightEnd(logEntry: JsObject): Unit = {
+    try {
+      require("Trigger rule .+ end".r.matches(getMessage(logEntry)))
+
+      metricCountData(getSpanId(logEntry)) += ("in-flight-end" -> getInFlight(logEntry))
+    } catch {
+      case NonFatal(_) =>
+    }
+  }
+
+  private def addSteps(logEntry: JsObject): Unit = {
+    try {
+      val count = getMetrics(logEntry)
+        .fields("steps")
+        .expect(
+          "JsNumber",
+          { case JsNumber(value) =>
+            value.longValue
+          },
+        )
+        .orConverterException
+
+      metricCountData(getSpanParentId(logEntry)) += ("steps" -> count)
+    } catch {
+      case NonFatal(_) =>
+    }
+  }
+
+  private def addRuleEvaluation(logEntry: JsObject): Unit = {
+    try {
+      val timing = getMetrics(logEntry)
+        .fields("duration")
+        .asJsObject
+        .fields("rule-evaluation")
+        .expect(
+          "JsString",
+          { case JsString(value) =>
+            Duration(value.replace('u', 'µ')).asInstanceOf[FiniteDuration]
```

Comment:
Prefer to use a UTF8 representation of `µ` and so avoid any issues with logging or printing this character